### DISCUSSION TITLE
Consider spring boot minor upgrade as major

### DISCUSSION THREAD

**Discussion by dioniseo** [renovatebot/renovate#10121] [Request Help] [answered]
Hello, I'm interesting if its possible somehow to let Renovate Bot know that we want to identify minor upgrade of Spring Boot (e.g 2.4.5 -> 2.5.0) not as a minor version upgrade but as a major upgrade (usually such minor upgrades of spring boot may cause non compatible changes that needs some refactoring of the code to be able to use newer versions so we want it to be considered as a major upgrade).

After documentation review looks like in some cases we can configure allowedVersion parameter as regexp with named groups, smth like this:

`"allowedVersions": "/^(?<major>\\d+)\\.(?<minor>\\d+)\\.(?<patch>\\d+)?$/"`

Does renovate bot use these named groups during upgrade process to understand what version is major/minor/patch? Is it possible to use combined versions like 2.5 as a major group and 3rd digit as a minor group?

**Comment by rarkins** [comment, marked as answer]
> Does renovate bot use these named groups during upgrade process to understand what version is major/minor/patch?

You need to check out [`regex` versioning](https://docs.renovatebot.com/modules/versioning/#regular-expression-versioning) instead of `allowedVersions`. In that case it behaves as I think you expect. Something like this:

```json
{
  "packageRules": [{
    "matchPackagePrefixes": ["org.springframework.boot:"],
     "versioning": "regex:^2\\.(?<major>\\d+)(\\.(?<minor>\\d+))$"
  }] 
}
```

**Reply by dioniseo** [reply]
Cool, thanks!
